tiledb/sm/compressors/rle_compressor.h

```diff
@@ -73,8 +78,104 @@ class RLE {
 
   /** Returns the compression overhead for the given input. */
   static uint64_t overhead(uint64_t nbytes, uint64_t value_size);
-};
 
+  /**
+   * Compress variable sized strings in contiguous memory to RLE format
+   *
+   * @tparam T Type of integer to store run legths, must fit max num of
+   * repetitions in input
+   * @tparam P Type of integer to store string sizes, must fit max length input
+   * string
```

Comment:
C++ specifically says that `size_t` is a type that can represent the size of the largest objects that can fit in memory. It seems that these type parameters are extraneous at best.

Similarly, using `uint64_t` for byte sizes elsewhere isn't correct, strictly speaking. This is nowhere near the only place where this issue arises, but we don't need to be making the problem worse.

We do need to use `uint64_t` for lengths that don't need to fit into memory, notably file sizes. This is all to say that we don't have a good way of dealing with this issue yet. We need at least another size type defined (perhaps more than one) and a convention for dealing with coherently.